File: mirror-reflection.py
class Solution(object):
    def mirrorReflection(self, p, q):
        while p % 2 == 0 and q % 2 == 0: p, q =  p / 2, q / 2
        if p % 2 == 0:
            return 2;
        elif q % 2 == 0:
            return 0;
        else:
            return 1;

s = Solution()
x = s.mirrorReflection(2, 1)
print(x)

File: cpp/mirror-reflection.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    int mirrorReflection(int p, int q) {
        while (p % 2 == 0 && q % 2 == 0)  p >>= 1, q >>= 1;
        if (p % 2 == 0)
            return 2;
        else if (q % 2 == 0)
            return 0;
        else
            return 1;
    }
};

int main(int argc, const char* argv[])
{
    Solution s;
    cout << s.mirrorReflection(2, 1);
    cout << s.mirrorReflection(5, 2);
}

